src/Infrastructure/Injector/SimpleInjector.php

```diff
@@ -377,7 +377,14 @@ private function resolve_argument_by_name(
 		// Check if we have mapped this argument for the specific class.
 		if ( \array_key_exists( $class, $this->argument_mappings )
 			&& \array_key_exists( $name, $this->argument_mappings[ $class ] ) ) {
-			return $this->argument_mappings[ $class ][ $name ];
+			$value = $this->argument_mappings[ $class ][ $name ];
+
+			// Closures are immediately resolved, to provide lazy resolution.
+			if ( is_callable( $value ) ) {
```

Comment:
If this is specifically for closures, should this not be?

```suggestion
			if ( $value instanceof Closure ) {
```

Reply:
No, that would require actual Closure objects to be instantiated to make use of this.

Reply:
How?

```
$x = function () {};
var_dump( $x instanceof Closure );
// => true
```

Reply:
It's problematic as soon as you're more WordPress-ish: https://3v4l.org/bP3FI

Reply:
Right. I was just confused because the comment specifically references closures. So what the comment should say is callables/callbacks.